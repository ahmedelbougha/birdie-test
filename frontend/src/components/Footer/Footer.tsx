import { StyledFooter, StyledFooterLogo } from '.';
import { Container } from '../styles';

export function Footer() {
  return (
    <StyledFooter>
      <Container>
        <StyledFooterLogo src="/images/birdie-logo.svg" width="110" />
      </Container>
    </StyledFooter>
  );
}
