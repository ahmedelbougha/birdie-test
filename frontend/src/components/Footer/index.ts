export { Footer as default } from "./Footer";
export {
    StyledFooter,
    StyledFooterLogo
} from "./Footer.styled";

