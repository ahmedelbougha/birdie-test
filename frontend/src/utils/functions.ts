import axios, { AxiosResponse } from "axios";
import { EventTypes } from './constants';


/**
 * General API Requester
 * @returns Promise<AxiosResponse<any, any>>
 */
 export function apiRequest(path: string): Promise<AxiosResponse<any, any>> {
  if (path.length === 0) {
    throw new Error("Request path cannot be empty!");
  }
  return axios.request({
    method: "GET",
    url: `${process.env.REACT_APP_BACKEND_URL}/${path}`,
  });
}


export function buildEventTableEvents() {
  let eventTypes = [];

  // get the main keys of event types in order to use them to display events counts
  // in the EventTable
  const eventTypeKeys = Object.keys(EventTypes);

  for (let i = 0; i < eventTypeKeys.length; i += 2) {
    // build eventTypes array with 2 elements at a time
    // to build the EventTable as 2 events with their count per row
    eventTypes.push([eventTypeKeys[i], eventTypeKeys[i + 1]]);
  }

  return eventTypes;
}
