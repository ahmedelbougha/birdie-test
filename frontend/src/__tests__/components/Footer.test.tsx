import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Footer from '../../components/Footer';
import { theme } from '../../utils/constants';

describe('Footer', () => {
  it('renders', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Footer />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
