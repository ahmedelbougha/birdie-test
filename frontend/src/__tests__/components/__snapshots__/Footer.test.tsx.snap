// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<footer
  className="sc-jSUZER jzYyze"
>
  <div
    className="sc-gswNZR lhJxqP"
  >
    <img
      className="sc-gKPRtg VGwCa"
      src="/images/birdie-logo.svg"
      width="110"
    />
  </div>
</footer>
`;
